src/tools/android/java/com/google/devtools/build/android/ManifestMergerAction.java

```diff
@@ -231,11 +231,6 @@ public static void main(String[] args) throws Exception {
         // to the expected location of the output.
         Files.copy(manifest, options.manifestOutput, StandardCopyOption.REPLACE_EXISTING);
       }
-    } catch (AndroidManifestProcessor.ManifestProcessingException e) {
```

Comment:
The root `Exception` can catch everything, minus the `System.exit`.

Reply:
This was explicitly added to fix https://github.com/bazelbuild/bazel/issues/6904. We should not unfix that.

Reply:
Thoughts on raising this specific logic up into `ResourceProcessorBusyBox` where we can log the exception and return the proper exit code inside `processRequest`?

Looking at some of the other options, they don't really

Reply:
@ahumesky

Reply:
Well, this is a bit strange. There are actually two exception classes with this name:
https://github.com/bazelbuild/bazel/blob/09c621e4cf5b968f4c6cdf905ab142d5961f9ddc/src/tools/android/java/com/google/devtools/build/android/ManifestProcessingException.java
https://github.com/bazelbuild/bazel/blob/09c621e4cf5b968f4c6cdf905ab142d5961f9ddc/src/tools/android/java/com/google/devtools/build/android/AndroidManifestProcessor.java#L60

And they're even both used in the same class:
https://github.com/bazelbuild/bazel/blob/09c621e4cf5b968f4c6cdf905ab142d5961f9ddc/src/tools/android/java/com/google/devtools/build/android/AndroidManifest.java#L85
https://github.com/bazelbuild/bazel/blob/09c621e4cf5b968f4c6cdf905ab142d5961f9ddc/src/tools/android/java/com/google/devtools/build/android/AndroidManifest.java#L145

We should converge onto one to make this less confusing....

Looks like the top-level class is also being used to capture two different kinds of exceptions:
https://github.com/bazelbuild/bazel/blob/09c621e4cf5b968f4c6cdf905ab142d5961f9ddc/src/tools/android/java/com/google/devtools/build/android/AndroidManifest.java#L84-L86

IOException is one thing, but XMLStreamException might be from badly formed xml...

Either way, if we just throw here, this exception will get caught by the `catch (Exception e)` in ResourceProcessorBusyBox, and do the same basic logging. So probably don't need to make ResourceProcessorBusyBox aware of lower level exceptions.

Reply:
> Either way, if we just throw here, this exception will get caught by the catch (Exception e) in ResourceProcessorBusyBox, and do the same basic logging. So probably don't need to make ResourceProcessorBusyBox aware of lower level exceptions.

The generic `catch (Exception e) {` in `ResourceProcessorBusyBox` will re-throw the exception after logging the error and bubble that up as an exception coming from `main`, while `catch (AndroidManifestProcessor.ManifestProcessingException e) {`  gracefully handles the exception with an exit code of `1` being returned directly via the work response (Prior to this PR is was being returned as a `System#exit` with 1). Is this slight behavioral change fine to merge?

Reply:
Probably isn't a big deal given that this PR wraps all unhandled exceptions with a proper exit code of 1 in order to confirm to `WorkRequestHandler`.

https://github.com/bazelbuild/bazel/pull/14428/files#diff-6bbb0aaf4cf2d425734021815617e2b6203223abe7dc76ca8030a90e9f043873R213

Reply:
@larsrc-google is this in a mergable state so that https://github.com/bazelbuild/bazel/pull/14428/ is unblocked?

Reply:
Sorry, I'm catching up on this thread right now -- I'll probably add in a comment in the ManifestProcessingException case handler to explain why its handler is almost the same as the generic case, minus the logging statement. I think it should be fine -- either we have essentially two debug log lines of the same error message, or we have a slightly funky-looking try/catch block. Cutting down on noise in the console seems preferable in this case.